### PR TITLE
Backend game logic

### PR DESCRIPTION
**Pull request by UntidyRAM**
# Description
Fixed a logic error that let a player join a game that was already in progress.
Refactored the backend code after fixing the logic error.
Added tests for the backend code.

Related #55

### CODE REVIEW

**Review by Ray-R4y — APPROVED** [review]
Design
Good design on the connections, multiple lobbys are enabled

Functionality
Code functions well with functioning tests

Complexity
code is simple to understand with comment where needed

Tests
Many tests were implemented to test the functionality of the code

Naming
Naming of the variables follows the camelcase naming style.

Comments
the comments are good and simple to understand

Style
The style follows the google javascript style guide.

Consistency
The code is consistent with the style guide

Documentation
All Documentations exist

Error handling 
Errors were handled well

Good things
code is easy to understand , and the tests implemented are functional. the code functions correctly and good design is implemented.

File: src/public/game.js [hunk 1]
```diff
@@ -136,8 +136,10 @@ function updateWordleTableColor () {
   }
 }
 
-function updateOpponentColors (arrayOfColors) {
-  const table = document.getElementById('opponentColorTable')
+function updateOpponentColors (arrayOfColors, playerNum) {
+  let table = document.getElementById(`opponent${playerNum}Table`)
```
Comment (JessWhosBack) [inline src/public/game.js:140]: Great use of the playerNum variable to control the html

File: src/views/game.html [hunk 2]
```diff
@@ -28,48 +28,48 @@
...
   <div id="wordleDiv" class="unselectable">
-    <table id="wordleTable">
+    <table id="wordleTable" class="table">
       <tr>
-        <th></th>
-        <th></th>
-        <th></th>
-        <th></th>
-        <th></th>
+        <td></td>
```
Comment (JessWhosBack) [inline src/views/game.html:33]: Good catch, my bad for using th !

**Review by JessWhosBack — APPROVED** [review]
COMMENTS: Great comments as always
QUALITY: Good code, even picked up small errors of mine
NAMING CONVENTION AND STYLE: Standard js style and naming convention used consistently throughout

**Pull request merged**